generate-address-list.py

```diff
@@ -74,10 +87,31 @@ def write_addresses_json(addresses, asset, outpath):
     with open("{}/sanctioned_addresses_{}.json".format(outpath, asset), 'w') as out:
         out.write(json.dumps(addresses, indent=2)+"\n")
 
+def compute_sha256(file_path):
+    sha256_hash = hashlib.sha256()
+    with open(file_path, "rb") as f:
+        for chunk in iter(lambda: f.read(4096), b""):
+            sha256_hash.update(chunk)
+    return sha256_hash.hexdigest()
+
+def write_checksum_file(sha256, checksum_file_path):
+    with open(checksum_file_path, "w") as checksum_file:
+        checksum_file.write(f"SHA256({file_path}) = {sha256}\n")
 
 def main():
     args = parse_arguments()
 
+    # First, check if the checksum matches the one on the website
+    scraper = OfacWebsiteScraper()
+    try:
+        sha256_checksum_from_site = scraper.get_sha256_checksum()
+    finally:
+        scraper.close()
+
+    sha256_checksum_computed = compute_sha256(SDN_ADVANCED_FILE_PATH)
+    if sha256_checksum_computed != sha256_checksum_from_site:
+        raise "Checksums do not match."
```

Comment:
probably worth alerting with the bot on this scenario as well

Reply:
Should do so on an uncaught error